app/controllers/Application.scala

```diff
@@ -504,75 +520,87 @@ object Application extends Controller {
   }
 
   def renameNotebook(p: String) = Action(parse.tolerantJson) { request =>
-    val oldPath = URLDecoder.decode(p, UTF_8)
-    val newPath = (request.body \ "path").as[String]
-    Logger.info("RENAME → " + oldPath + " to " + newPath)
-    try {
-      val (newname, newpath) = notebookManager.rename(oldPath, newPath)
+    if (viewer){
+      BadRequest("Cannot rename notebook in viewer mode")
+    } else {
+      val oldPath = URLDecoder.decode(p, UTF_8)
```

Comment:
my only concern here, is it might be really hard to merge/maintain (as we have multiple branches, and some have modifications).
it would be better to rewrite this so it produces minimal diff. @andypetrella  I may take a look into this myself.